### PR TITLE
Add location into firebase

### PR DESCRIPTION
**Pull request by r-ismer**

### CODE REVIEW

File: app/build.gradle
Comment (koko-ng) [inline app/build.gradle:46]: Is this really mandatory? Android docs seems to tell it should really be avoided, moreover I don't see any strange tests that could justify it

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:10]: Where is it gone? :(

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfilePassions.kt:77]: Please, stay comments, we will really miss you.

File: app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsLocation.kt
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsLocation.kt:23]: I guess we are using Lausanne and EPFL as default values? Maybe we should merge them somehow, seems a little bit odd to me to have to different defaults.

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:27]: It is not very clear, why is the changed? Is `matches` now a list of IDs? If it is the case, I think renaming to something like matchesIds could be clearer.

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:202]: I think it is a bit risky to do a unsafe cast here, imagine a user put an invalid list of Strings in the db, could they make every app that encounters them while swiping crash? Maybe a safe cast is the way to go: [https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator](https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator)

**Review by koko-ng — CHANGES_REQUESTED** [review]
Overall it seems nice, despite some small points. I also thinks that your linter went a bit crazy and deleted a lot of comment, let me know if it was intentional.

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (cmsigrist) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:79]: Maybe you should add checks on the input length (that it only contains two doubles)

File: app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsLocation.kt
Comment (cmsigrist) [inline app/src/main/java/ch/epfl/sdp/blindly/settings/SettingsLocation.kt:49]: I think as Tibor said that we don't need the LAUSANNE_LATLING, even more so if it's only used as a default value, maybe it would be better to have an empty list then?

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (cmsigrist) [reply app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:202]: I don't know, to me it makes more sense to check the inputs, so that no invalid inputs are put on the database in the first place

**Review by cmsigrist — COMMENTED** [review]
Great! It's indeed a lot better to have the latitude and longitude rather than a String
I'll approve once you've #BringBackTheComments:'(

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:27]: It was actually discussed in the meetings, this is preferred not to unnecessarily store whole user for each match where as our matching algorithm only needs UIDs which are stored as Strings.

File: app/build.gradle
Comment (r-ismer) [reply app/build.gradle:46]: I had to because the unit tests won't pass without it

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (koko-ng) [reply app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:202]: Of couse but a malicious user could still send queries from outside the app to put invalid data in our db, especially as we aren't doing any kind of server-side validation of the data.

File: app/build.gradle
Comment (r-ismer) [reply app/build.gradle:46]: i checked on android dev https://developer.android.com/training/testing/unit-testing/local-unit-tests.html

**Review by koko-ng — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:12]: I'm not really sure of this naming, what is the point of having the constant :D Maybe something like SIZE_LAT_AND_LNG ? I leave it up to you.

**Review by cmsigrist — APPROVED** [review]
Nice work! :)

**Review by cgntnr — APPROVED** [review]
A small remark on a constant otherwise the other comments should be enough. Thanks!

**Pull request merged**
